engine/util/src/main/scala/pl/touk/nussknacker/engine/util/typing/JsonToTypedMapConverter.scala

```diff
@@ -0,0 +1,86 @@
+package pl.touk.nussknacker.engine.util.typing
+
+import com.typesafe.scalalogging.LazyLogging
+import io.circe.{Json, JsonObject}
+import org.everit.json.schema.{ArraySchema, BooleanSchema, FalseSchema, NullSchema, NumberSchema, ObjectSchema, Schema, StringSchema, TrueSchema}
+import pl.touk.nussknacker.engine.api.typed.TypedMap
+import pl.touk.nussknacker.engine.api.typed.typing.{Typed, TypedObjectTypingResult, TypingResult}
+
+import scala.collection.JavaConverters._
+
+object JsonToTypedMapConverter {
+
+  def jsonToTypedMap(json: Json): TypedMap =
+    jsonObjectToTypedMap(json.asObject.getOrElse(JsonObject.empty))
+
+  private def jsonToAny(json: Json): Any = {
+    json.fold(
+      jsonNull = null,
+      jsonBoolean = identity,
+      jsonNumber = d => {
+        //circe 0.14.1 still looses precision as was in https://github.com/circe/circe/issues/1101.
```

Comment:
Does it? The issue seems resolved for quite a long time?

Reply:
In NK we are using circe `0.14.1` and still `JsonToTypedMapConverterTest` fails if we use `d.toBigDecimal`.
You can observe it by looking at the output for `.toBigDecimal` in this tests.
I was sure it was fixed but sadly I observe different behavior.